library/src/main/java/com/novoda/downloadmanager/lib/DownloadNotifier.java

```diff
@@ -167,27 +168,27 @@ private boolean shouldShowCompletedItem(int visibility) {
                 || visibility == NotificationVisibility.ACTIVE_OR_COMPLETE;
     }
 
-    private void addBatchToCluster(String tag, Map<String, List<DownloadBatch>> cluster, DownloadBatch batch) {
+    private void addBatchToCluster(String tag, Map<String, Collection<DownloadBatch>> cluster, DownloadBatch batch) {
         if (tag == null) {
             return;
         }
 
-        List<DownloadBatch> batches;
+        Collection<DownloadBatch> batches;
 
         if (cluster.containsKey(tag)) {
             batches = cluster.get(tag);
         } else {
-            batches = new ArrayList<>();
+            batches = new HashSet<>();
             cluster.put(tag, batches); // TODO not sure if this is right compared to ArrayListMultiMap
```

Comment:
@ouchadam uh? This comments are about this line `cluster.put(tag, batches); // TODO not sure if this is right compared to ArrayListMultiMap` :P